addons/portal/static/src/js/portal.js

```diff
@@ -53,6 +53,27 @@ publicWidget.registry.portalDetails = publicWidget.Widget.extend({
     },
 });
 
+publicWidget.registry.portalDetails = publicWidget.Widget.extend({
+    selector: '.o_portal_details',
+    events: {
+        'click .js_delete_address': '_onClickDeleteAddress',
+    },
+
+    _onClickDeleteAddress: function (ev) {
+        ev.preventDefault();
+        const confirmCallback = () => {
+            let form = $(ev.currentTarget.form);
+            let input = $('<input>').attr({
+                type: "hidden",
+                name: "address_action",
+                value: "delete"
+            });
+            form.append(input).submit();
```

Comment:
Seems overkill to submit a whole form for this.
Could we call the unlink method of the address model via rpc instead?
Not sure if portal users can do that, to double check.